src/allotropy/parsers/perkin_elmer_envision/perkin_elmer_envision_structure.py

```diff
@@ -52,69 +53,212 @@ def num_to_chars(n: int) -> str:
 class PlateInfo:
     number: str
     barcode: str
-    emission_filter_id: str
     measurement_time: Optional[str]
     measured_height: Optional[float]
     chamber_temperature_at_start: Optional[float]
 
     @staticmethod
-    def create(reader: CsvReader) -> Optional[PlateInfo]:
+    def get_series(reader: CsvReader) -> pd.Series[str]:
         assert_not_none(
             reader.pop_if_match("^Plate information"),
             msg="Unable to find expected plate information",
         )
 
-        series = df_to_series(
+        return df_to_series(
             assert_not_none(
                 reader.pop_csv_block_as_df(),
                 "Plate information CSV block",
             )
         ).replace(np.nan, None)
 
-        plate_number = try_str_from_series(
+    @staticmethod
+    def get_plate_number(series: pd.Series[str]) -> str:
+        return try_str_from_series(
             series,
             "Plate",
             msg="Unable to find plate number",
         )
 
-        optional_barcode = try_str_from_series_or_none(series, "Barcode")
-        raw_barcode = (optional_barcode or '=""').removeprefix('="').removesuffix('"')
-        barcode = raw_barcode or f"Plate {plate_number}"
+    @staticmethod
+    def get_barcode(series: pd.Series[str], plate_number: str) -> str:
+        raw_barcode = try_str_from_series_or_none(series, "Barcode")
+        barcode = (raw_barcode or '=""').removeprefix('="').removesuffix('"')
+        return barcode or f"Plate {plate_number}"
+
+
+@dataclass
+class CalculatedPlateInfo(PlateInfo):
+    formula: str
+    name: str
+
+    @staticmethod
+    def create(series: pd.Series[str]) -> CalculatedPlateInfo:
+        plate_number = PlateInfo.get_plate_number(series)
+
+        formula = try_str_from_series(
+            series,
+            "Formula",
+            msg="Unable to get expected formula for calculated results section",
```

Comment:
would you mind changing all the error messages that you've added throughout this PR? I changed the guidelines recently https://github.com/Benchling-Open-Source/allotropy/blob/main/CONTRIBUTING.md#error-messaging (in particular, changing `get` to `find` here and in other places, ending with a period, and maybe printing the `series` as well if you think that would be valuable to an end user)